cpp/src/parquet/statistics.cc

```diff
@@ -648,6 +664,16 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     this->num_values_ = 0;
   }
 
+  void ResetHasFlags() {
+    this->has_min_max_ = false;
+    // writer will write `distinct_count` to EncodedStatistics.
+    // So, disable it when reset.
+    this->has_distinct_count_ = false;
+    // writer will always write `null_count` to EncodedStatistics.
+    // So, enable it when reset.
+    this->has_null_count_ = true;
```

Comment:
```suggestion
    // has_min_max_ will only be set when it meets any valid value.
    this->has_min_max_ = false;
    // has_distinct_count_ will only be set once SetDistinctCount()
    // is called because distinct count calculation is not cheap and
    // disabled by default.
    this->has_distinct_count_ = false;
    // Null count calculation is cheap and enabled by default.
    this->has_null_count_ = true;
```